packaging/linux/flatpak/dev.lizardbyte.sunshine.yml

```diff
@@ -317,6 +317,7 @@ modules:
       - cd ${FLATPAK_BUILDER_BUILDDIR} && npm install
     config-opts:
       - -DCMAKE_BUILD_TYPE=Release
+      - -DCMAKE_INSTALL_DATAROOTDIR=/app/share
```

Comment:
I think this needs to be added to the following locations, unless the default is okay. Please confirm.

- https://github.com/LizardByte/Sunshine/blob/11d472362cb1f5896c8a2d3de4b2184a2f0a37ad/.github/workflows/CI.yml#L402-L412
- https://github.com/LizardByte/Sunshine/blob/11d472362cb1f5896c8a2d3de4b2184a2f0a37ad/packaging/linux/Arch/PKGBUILD#L80-L86
- https://github.com/LizardByte/Sunshine/blob/11d472362cb1f5896c8a2d3de4b2184a2f0a37ad/docker/debian-bookworm.dockerfile#L105-L115
- https://github.com/LizardByte/Sunshine/blob/11d472362cb1f5896c8a2d3de4b2184a2f0a37ad/docker/debian-bullseye.dockerfile#L119-L129
- https://github.com/LizardByte/Sunshine/blob/11d472362cb1f5896c8a2d3de4b2184a2f0a37ad/docker/fedora-38.dockerfile#L106-L115
- https://github.com/LizardByte/Sunshine/blob/11d472362cb1f5896c8a2d3de4b2184a2f0a37ad/docker/fedora-39.dockerfile#L106-L115
- https://github.com/LizardByte/Sunshine/blob/11d472362cb1f5896c8a2d3de4b2184a2f0a37ad/docker/ubuntu-20.04.dockerfile#L155-L165
- https://github.com/LizardByte/Sunshine/blob/11d472362cb1f5896c8a2d3de4b2184a2f0a37ad/docker/ubuntu-22.04.dockerfile#L120-L130

Reply:
For regular Linux distributions the defaults which CMake with the use of GNUInstallDirs figures out **should** be okay, as thats main purpose of it.  (See also the comment from @Conan-Kudo above in regards to lib vs lib32/lib64 handling, which for example we don't have on Exherbo as we set different arch-specific prefixes instead.)

Reply:
Thinking about it it's probably also not required for the flatpak as well as it sets  `-DCMAKE_INSTALL_PREFIX=/app` so stuff should properly end up being installed there.

Reply:
> Probably at the top of `./cmake/packaging/unix.cmake`

@tgurr @ReenigneArcher `special_package_configuration.cmake` is include before `unix.cmake` been include in CMakeLists.txt, therefore `GNUInstallDirs` is not enabled when configure desktop file, the `@CMAKE_INSTALL_DATAROOTDIR@` in `.desktop` files is invalid and empty string, this broken both regular and flatpak version of the desktop file in v0.22.0 release. https://github.com/LizardByte/Sunshine/blob/faeeb7ec573590c798111024510b2d46c03beaeb/packaging/linux/sunshine.desktop#L15 but when I try to add `include(GNUInstallDirs)` to `special_package_configuration.cmake` and generate the desktop file again,
```
Exec=gio launch share/applications/sunshine_terminal.desktop
```
the `/usr/` in path is missing, a cmake bug? my distro is archlinux.
so how to fix this, revert the change for these desktop file, add `-DCMAKE_INSTALL_DATAROOTDIR="/usr/share"` in PKGBUILD and anywhere needed, or wait cmake fix `@CMAKE_INSTALL_DATAROOTDIR@` in `configure_file()`?

Reply:
Adding `-DCMAKE_INSTALL_DATAROOTDIR:PATH=/usr/share` where needed appears to be the correct approach to me, e.g. on Exherbo we have to do that anyways because we need to install arch-independent stuff into /usr and arch-dependent stuff into prefix e.g. /usr/x86_64-pc-linux-gnu like bin, lib and so on which was the reason for this MR in the first place. Unfortunate this doesn't work automatically for `configure_file()` but I can see no reason why `@CMAKE_INSTALL_PREFIX@` would work but `@CMAKE_INSTALL_DATAROOTDIR@` would not, according the the PKGBUILD `@CMAKE_INSTALL_PREFIX@` was and is also manually set: https://github.com/LizardByte/Sunshine/blob/nightly/packaging/linux/Arch/PKGBUILD#L65 which might explain why things worked in the first place and brings me back to my suggestion of passing `-DCMAKE_INSTALL_DATAROOTDIR:PATH=/usr/share` when/where needed. Hope you all agree.

Reply:
@tgurr wait according to cmake doc you post above, variable `CMAKE_INSTALL_DATAROOT`'s value should really just `share`.
> `CMAKE_INSTALL_FULL_<dir>`
>
>    The absolute path generated from the corresponding `CMAKE_INSTALL_<dir>` value. If the value is not already an absolute path, an absolute path is constructed typically by prepending the value of the [CMAKE_INSTALL_PREFIX](https://cmake.org/cmake/help/v3.18/variable/CMAKE_INSTALL_PREFIX.html#variable:CMAKE_INSTALL_PREFIX) variable. However, there are some [special cases](https://cmake.org/cmake/help/v3.18/module/GNUInstallDirs.html#special-cases) as documented below.

 you need `CMAKE_INSTALL_FULL_DATAROOT` for value that auto prepend `CMAKE_INSTALL_PREFIX` in to `CMAKE_INSTALL_DATAROOT`, maybe you can make a new MR for update this and add `include(GNUInstallDirs)` in `special_package_configuration.cmake`?

Reply:
@detiam Do I get you right that you mean:

```
-Exec=gio launch @CMAKE_INSTALL_DATAROOTDIR@/applications/sunshine_terminal.desktop
+Exec=gio launch @CMAKE_INSTALL_FULL_DATAROOTDIR@/applications/sunshine_terminal.desktop
```
I lack the possibility to test your usecase and also don't want to steal the credits from you so if you'd prepare an MR instead I'd happiliy test my usecase to see if something breaks for me (which I don't expect, in the worst case we just set the different variable manually for our build). If you still want to me to prepare an MR instead please advise.

Reply:
> @detiam Do I get you right that you mean:
> 
> ```
> -Exec=gio launch @CMAKE_INSTALL_DATAROOTDIR@/applications/sunshine_terminal.desktop
> +Exec=gio launch @CMAKE_INSTALL_FULL_DATAROOTDIR@/applications/sunshine_terminal.desktop
> ```
> 
> I lack the possibility to test your usecase and also don't want to steal the credits from you so if you'd prepare an MR instead I'd happiliy test my usecase to see if something breaks for me (which I don't expect, in the worst case we just set the different variable manually for our build). If you still want to me to prepare an MR instead please advise.

ok I opened a new PR here #2223 :)